core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3054,6 +3054,26 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleAlterIsrRequest(request: RequestChannel.Request): Unit = {
+    val alterIsrRequest = request.body[AlterIsrRequest]
+
+    if (authorize(request.context, CLUSTER_ACTION, CLUSTER, CLUSTER_NAME)) {
```

Comment:
nit: it's subjective, so feel free to ignore, but I find this a little easier to read if we handle the error cases first. So..
```scala
if (!authorize(request.context, CLUSTER_ACTION, CLUSTER, CLUSTER_NAME)) {
      sendResponseMaybeThrottle(request, requestThrottleMs =>
        alterIsrRequest.getErrorResponse(requestThrottleMs, Errors.CLUSTER_AUTHORIZATION_FAILED.exception))Errors.CLUSTER_AUTHORIZATION_FAILED.exception))  
} else if (!controller.isActive) {
     sendResponseMaybeThrottle(request, requestThrottleMs =>
        alterIsrRequest.getErrorResponse(requestThrottleMs, Errors.NOT_CONTROLLER.exception()))
} else {
...
}
```
Basically we're discarding the error cases so that the successful path continues flowing downward and we're avoiding extra nesting. Like I said, it's subjective.